kanidmd/src/lib/be/idl_sqlite.rs

```diff
@@ -997,6 +1002,94 @@ impl IdlSqliteWriteTransaction {
         })
     }
 
+    pub fn store_idx_slope_analysis(
+        &self,
+        audit: &mut AuditScope,
+        slopes: &HashMap<IdxKey, IdxSlope>,
+    ) -> Result<(), OperationError> {
+        self.conn
+            .execute(
+                "CREATE TABLE IF NOT EXISTS idxslope_analysis (
+                    id TEXT PRIMARY KEY,
+                    slope INTEGER
+                )",
+                [],
+            )
+            .map(|_| ())
+            .map_err(|e| {
+                ladmin_error!(audit, "SQLite Error {:?}", e);
+                OperationError::SqliteError
+            })?;
+
+        // Remove any data if it exists.
+        self.conn
+            .execute("DELETE FROM idxslope_analysis", [])
+            .map(|_| ())
+            .map_err(|e| {
+                ladmin_error!(audit, "SQLite Error {:?}", e);
+                OperationError::SqliteError
+            })?;
+
+        slopes.iter().try_for_each(|(k, v)| {
+            let key = format!("idx_{}_{}", k.itype.as_idx_str(), k.attr);
+            self.conn
+                .execute(
+                    "INSERT OR REPLACE INTO idxslope_analysis (id, slope) VALUES(:id, :slope)",
+                    named_params! {
+                        ":id": &key,
+                        ":slope": &v,
+                    },
+                )
+                .map(|_| ())
+                .map_err(|e| {
+                    eprintln!("CRITICAL: rusqlite error {:?}", e);
```

Comment:
How much should the admin be shitting their pants at exactly this moment that something has gone wrong :)

Reply:
Would it be a good idea to add a feature in https://github.com/kanidm/kanidm/pull/497 so that regular logs go to a file, but we could have a special `alarm!` macro that goes through the subscriber and is written to a log but also immediately outputted to `stderr` or wherever is most appropriate/visible?

Reply:
Yes, that's a very good idea, and you should definitely add that to the list of requirements/nice to have :)